File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
import java.util.List;
import java.util.ArrayList;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = {7,7};
        List<Integer> list = new ArrayList<> ();
        
        for(int i : win_nums){
            list.add(i);
        }
        
        for(int i=0; i<6; i++){
            if(lottos[i]==0){
                answer[0]--;
            }else if(list.contains(lottos[i])){
                answer[0]--;
                answer[1]--;
            }
        }
        
        if(answer[0]==7) answer[0]--;
        if(answer[1]==7) answer[1]--;
        
        return answer;
    }
}

File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
    const mystery = lottos.filter(n => n===0).length;
    const correct = lottos.reduce((cnt, n) => win_nums.includes(n)? cnt+1: cnt, 0);
    return [7 - ((correct+mystery) || 1), 7 - (correct || 1)];
}